src/deriv_api/__tests__/DerivAPIBasic.js

```diff
@@ -22,6 +22,13 @@ test('API construction with endpoint', () => {
     expect(() => new DerivAPIBasic({ endpoint: 1, app_id: 12345 })).toThrow();
 });
 
+test('API should change endpoint', () => {
```

Comment:
This pretty much affects all of other tests, it's good if we can have a test case in all of the request and subscription related tests for a `changeSocket` call.